src/index.js

```diff
@@ -25,23 +25,66 @@ const { fetch, Response } = process.env.HELIX_FETCH_FORCE_HTTP1
   : fetchAPI;
 const commence = require('./stop');
 
+function getMdUrl(host, path, log) {
+  let mdPath;
+  const file = path.split('/').pop();
+  if (file.endsWith('.html')) {
+    mdPath = path.replace(/\.html$/, '.md');
+  } else if (!file.contains('.')) {
+    mdPath = `${path}.md`;
+  }
+  if (!mdPath) {
+    log.debug('not an html document, so no markdown purging required');
+    return null;
+  }
+  const ghDetails = host.split('.')[0].split('--');
+  if (ghDetails.length < 2) {
+    log.warn('invalid inner cdn url');
+    return null;
+  }
+  const owner = ghDetails.pop();
+  const repo = ghDetails.pop();
+  const branch = ghDetails[0] || 'master';
+  return `https://${branch}--${repo}--${owner}.hlx.page${mdPath}`;
+}
+
 async function purgeInner(host, path, service, token, log) {
+  const results = [];
+  const mdUrl = getMdUrl(host, path, log);
+  if (mdUrl) {
+    try {
+      const res = await fetch(mdUrl, {
+        method: 'PURGE',
+      });
+      const msg = await res.text();
+      log.debug(msg);
+      if (!res.ok) {
+        throw new Error(msg);
```

Comment:
do you really want to abort here? why not add an error result like below?

Reply:
The code doesn't abort here, as the error is caught below, along with any possible error thrown by `fetch.